File: Assignment 1/SelectionSort/SelectionSort.py
import sys
from time import time


def SelectionSort(arr):
    temp = 0
    arr2 = arr.copy()
    for i in range(len(arr2)):
        minimo = i
        for j in range(i+1,len(arr2)):
            if arr2[minimo] > arr2[j]:
                minimo = j
        temp = arr2[i]
        arr2[i] = arr2[minimo]
        arr2[minimo] = temp

#input de datos
archivo = sys.stdin.readlines()
n = int(archivo[0])
arr = []
dato = archivo[1].split()
for i in range (n):
    arr.append(int(dato[i]))

#medir tiempos
start = time()
SelectionSort(arr)
finish = time()

print(finish-start,end = '')

File: Assignment 1/SelectionSort/SelectionSort.cpp
#include <iostream>
#include <ctime>
using namespace std;

void selectionSort(int *arr,int size)
{
    int temp;
    for(int i = 0 ; i < size - 1 ;  i++ )
    {
        int minimo = i;
        for ( int j = i+1 ; j < size  ;  j++ )
        {
            if( arr[minimo] > arr[j] )
                minimo = j ;
        }

        temp = arr[i];
        arr[i] = arr[minimo];
        arr[minimo] = temp;
        
    }
}



int main()
{
    int input,size;
    cin >> size;
    int array[size];
    for(int i = 0 ; i < size ; i++ )
    {
        cin >> input;
        array[i] = input;
    }
    
    unsigned t0,t1;

    t0 = clock();
    selectionSort(array,size);
    t1 = clock();

    double time = (double(t1-t0)/CLOCKS_PER_SEC);
    cout << time << endl;

    return 0;
}
